backend/layers/persistence/persistence.py

```diff
@@ -257,6 +258,25 @@ def _get_datasets(self, ids: List[DatasetVersionId]) -> List[DatasetVersion]:
                 datasets.append(self._row_to_dataset_version(version, canonical_dataset, version_artifacts))
         return datasets
 
+    @staticmethod
+    def _set_revised_at_field(
+        dataset_versions: List[DatasetVersion], collection_versions: List[CollectionVersion]
+    ) -> None:
+        """
+        Sets the `revised_at` field on the CanonicalDataset object for each DatasetVersion object
+        """
+        for dataset_version in dataset_versions:
+            try:
+                version_published_at, collection_version_id = get_published_at_and_collection_version_id_else_not_found(
+                    dataset_version, collection_versions
+                )
+                if version_published_at > dataset_version.canonical_dataset.published_at:
```

Comment:
We should add None checks here, to avoid an Exception on dataset_version.canonical_dataset.published_at is None or version_published_at is None

Reply:
the function that returns `version_published_at` can't return `None` for it, and in the case where a Dataset has never been published before, i.e. `dataset_version.canonical_dataset.published_at` is `None`, then a `DatasetVersionNotFoundException` is thrown, and in this case it is caught because the calling function wishes to simply skip over such cases.